File: objects/examples/personal.py
#!/usr/bin/env python
# vim: ft=python ts=4 sw=4 expandtab
#
# date: Fri Jul  8 20:34:23 2016

import os
import sys
import json
import re

def getinput(describe):
    sys.stderr.write(describe + ': ')
    line = sys.stdin.readline()
    line = line.rstrip()
    return line

def jsonfile():
    basefile = os.path.basename(__file__)
    return os.path.join(os.path.dirname(__file__),os.path.splitext(basefile)[0] + '.json')

def main(arguments):
    json_path = jsonfile()
    json_out = ''

    correct = False
    if not os.path.exists(json_path):
        while True:
            sys.stderr.write("Initial setup %s \n" % json_path)
            output = { 'full_name': getinput('Full Name'), 'username': getinput('Username'), 'email': getinput('Email Address') }
            json_out = json.dumps(output, indent = 4, separators = (',', ': '))

            sys.stderr.write(json_out + "\n")

            check = getinput("Does this look correct? (Yes/No)")

            if re.match('^(?:[Yy][Ee][Ss]|[Yy])',check):
                correct = True
                break

    if correct:
        with open(json_path,'w') as fd:
            fd.write(json_out)

    if os.path.exists(json_path):
        with open(json_path, 'r') as fd:
            lines = fd.readlines()
            for line in lines:
                sys.stdout.write(line)

if __name__ == '__main__':
    sys.exit(main(sys.argv[1:]))

File: objects/examples/personal.rb
#!/usr/bin/env ruby
# vim: ft=ruby ts=2 sw=2 expandtab
#
# date: Fri Jul  8 17:52:29 2016
# contact/creator: Dexter Plameras

require 'json'

#
# Definitions
#
def getinput(describe)
  STDERR.print("#{describe}: ")
  STDIN.gets.chomp
end

#
# Main
#
file = File.join(File.dirname($0), "#{File.basename($0, '.*')}.json")

# Check for object/personal.json
correct = false
unless File.exist?(file)
  output = nil
  begin
    STDERR.puts "Initial setup #{file}"
    output = {
        full_name: getinput('Full Name'),
        email: getinput('Email Address')
    }

    puts JSON.pretty_generate(output)
    check = getinput "Does this look correct? (Yes/No)"

    correct = true if check =~ /^(?:[Yy][Ee][Ss]|[Yy])$/
  end until correct
end

if correct
  File.open(file, 'w') do |file|
    file.puts(JSON.pretty_generate(output))
  end
end

if File.exist?(file)
  File.open(file, 'r') do |file|
    file.each_line do |line|
      print line
    end
  end
end

